gpu_support/add_nvidia_gpu_support.sh

```diff
@@ -0,0 +1,180 @@
+else
+  # - as an installation location just use $EESSI_SOFTWARE_PATH but replacing `versions` with `host_injections`
+  #   (CUDA is a binary installation so no need to worry too much about this)
+  # TODO: The install is pretty fat, you need lots of space for download/unpack/install (~3*5GB), need to do a space check before we proceed
+  avail_space=$(df --output=avail ${cuda_install_dir}/ | tail -n 1 | awk '{print $1}')
+  if (( ${avail_space} < 16000000 )); then
+    echo "Need more disk space to install CUDA, exiting now..."
+    exit 1
+  fi
+  # install cuda in host_injections
+  module load EasyBuild
+  eb --installpath=${cuda_install_dir}/ CUDA-${install_cuda_version}.eb
+  ret=$?
+  if [ $ret -ne 0 ]; then
+    echo "CUDA installation failed, please check EasyBuild logs..."
+    exit 1
+  fi
+fi
+
+# Check if the CUDA compat libraries are installed and compatible with the target CUDA version
+# if not find the latest version of the compatibility libraries and install them
+
+# get URL to latest CUDA compat libs, exit if URL is invalid
+cuda_compat_urls="$($(dirname "$BASH_SOURCE")/get_cuda_compatlibs.sh)"
+ret=$?
+if [ $ret -ne 0 ]; then
+  echo $cuda_compat_urls
+  exit 1
+fi
+
+# loop over the compat library versions until we get one that works for us
+keep_driver_check=1
+# Do a maximum of five attempts
+for value in {1..5}
+do
+    latest_cuda_compat_url=$(echo $cuda_compat_urls | cut -d " " -f1)
+    # Chomp that value out of the list
+    cuda_compat_urls=$(echo $cuda_compat_urls | cut -d " " -f2-)
+    latest_driver_version="${latest_cuda_compat_url%-*}"
+    latest_driver_version="${latest_driver_version##*-}"
+    # URLs differ for different OSes; check if we already have a number, if not remove string part that is not needed
+    if [[ ! $latest_driver_version =~ ^[0-9]+$ ]]; then
+      latest_driver_version="${latest_driver_version##*_}"
+    fi
+
+    install_compat_libs=false
+    host_injections_dir="/cvmfs/pilot.eessi-hpc.org/host_injections/nvidia"
+    # libcuda.so points to actual cuda compat lib with driver version in its name
+    # if this file exists, cuda compat libs are installed and we can compare the version
+    if [ -e $host_injections_dir/latest/compat/libcuda.so ]; then
+      eessi_driver_version=$( realpath $host_injections_dir/latest/compat/libcuda.so)
+      eessi_driver_version="${eessi_driver_version##*so.}"
+    else
+      eessi_driver_version=0
+    fi
+
+    if [ $keep_driver_check -eq 1 ]
+    then
+      # only keep the driver check for the latest version
+      keep_driver_check=0
+    else
+      eessi_driver_version=0
+    fi
+
+    if [ ${latest_driver_version//./} -gt ${eessi_driver_version//./} ]; then
+      install_compat_libs=true
+    else
+      echo "CUDA compat libs are up-to-date, skip installation."
+    fi
+
+    if [ "${install_compat_libs}" == true ]; then
+      source $(dirname "$BASH_SOURCE")/install_cuda_compatlibs.sh $latest_cuda_compat_url
```

Comment:
```suggestion
      bash $(dirname "$BASH_SOURCE")/install_cuda_compatlibs.sh $latest_cuda_compat_url
```
I think we don't need to `source` this anymore